app/controllers/concerns/lti_support.rb

```diff
@@ -53,24 +53,34 @@ def lti_provider
     end
 
     def user_from_lti
-      lti_user_id = params[:user_id]
-
-      # Match on both fields if possible
-      user = User.find_by(lms_user_id: lms_user_id, lti_user_id: lti_user_id)
+      # If the user is not logged in then params[:user_id] will be nil or empty string
+      if params[:user_id].present?
+        lti_user_id = params[:user_id]
+
+        # Match on both fields if possible
+        user = User.find_by(lms_user_id: lms_user_id, lti_user_id: lti_user_id)
+
+        # Match on only lms_user_id. This happens when a user uses OAuth to create and
+        # account before they ever do an LTI launch.
+        if user.blank? && lms_user_id.present?
+          if user = User.find_by(lms_user_id: lms_user_id)
+            if user.lti_user_id != lti_user_id
+              user.update!(lti_user_id: lti_user_id)
```

Comment:
There's a problem here if the lms_user_id changes but the lti_user_id doesn't.  This has caused issues in AA:

https://www.pivotaltracker.com/story/show/176195506

Why don't we match on `lti_user_id` first, since that's a unique key in the table,  and then on `lms_user_id`?  This will handle `lms_user_id` changing upstream.

Reply:
There is a condition where we create a user via the OAuth flow and the lms_user_id is present while the lti_user_id is empty.  This is meant to handle that case.